internal/testsupport/check.go

```diff
@@ -0,0 +1,96 @@
+package testsupport
+
+import (
+	"errors"
+	"fmt"
+	"strconv"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/terraform"
+	"github.com/hetznercloud/hcloud-go/hcloud"
+)
+
+// BackendKey allows to retrieve a resource from the backend using its ID.
+//
+// BackendKey must return true if the resource was found.
+type BackendKey func(c *hcloud.Client, id int) bool
```

Comment:
I would call it `HCloudKey`, as it is from the Hetzner Cloud. Backend is a bit too generic.